min_version_test/pubspec.yaml

```
@@ -18,10 +18,8 @@ publish_to: 'none' # Remove this line if you wish to publish to pub.dev
 version: 1.0.0+1
 
 environment:
-  sdk: '2.12.0'
-  flutter: '2.0.0'
-  # sdk: '>=2.12.0'
-  # flutter: '>=2.0.0'
+  sdk: '>=2.17.0 <3.0.0'
+  flutter: '>=3.0.0'
```

```suggestion
  sdk: '2.17.0'
  flutter: '3.0.0'
```

Keep the version pinned.